newIDE/app/src/ProjectsStorage/CloudStorageProvider/CloudProjectWriter.js

```diff
@@ -189,33 +190,40 @@ export const generateOnChooseSaveProjectAsLocation = ({
   fileMetadata: ?FileMetadata,
 |}): Promise<{|
   saveAsLocation: ?SaveAsLocation,
+  saveAsOptions: ?SaveAsOptions,
 |}> => {
   if (!authenticatedUser.authenticated) {
-    return { saveAsLocation: null };
+    return { saveAsLocation: null, saveAsOptions: null };
   }
 
-  const name = await new Promise(resolve => {
+  const options = await new Promise(resolve => {
     setDialog(() => (
-      <CloudSaveAsDialog
+      <SaveAsOptionsDialog
         onCancel={() => {
           closeDialog();
           resolve(null);
         }}
-        nameSuggestion={project.getName()}
-        onSave={(newName: string) => {
+        nameMaxLength={CLOUD_PROJECT_NAME_MAX_LENGTH}
+        nameSuggestion={
+          fileMetadata ? `${project.getName()} - Copy` : project.getName()
+        }
+        shouldAskForGameLinkRemoval={!!fileMetadata && !!fileMetadata.gameId}
```

Comment:
maybe name this prop `shouldAskForNewProjectUuid` to be consistent with the props you use in the component?